pandas/core/indexes/base.py

```diff
@@ -2441,6 +2454,13 @@ def intersection(self, other):
             taken.name = None
         return taken
 
+    def _create_empty_index(self, name):
```

Comment:
don't create new API, this is just ``self._shallow_copy([], name=name)``

Reply:
@jreback So the issue here is that `self._shallow_copy([], name=name)` doesn't work correctly for `RangeIndex`.  I can modify `RangeIndex._self_shallow_copy` to allow `[]` as an argument.  For an index of strings (objects) or an index of dtype `int64`,  `self._shallow_copy([], name=name)`  doesn't preserve the `dtype`.  And I haven't even looked at all the types yet.  So now the changes get more involved to handle this corner case of creating an empty index, which is why I created the `create_empty_index`.  The idea here is to have an empty index that preserves all the other attributes (dtype, categories, range step, etc.) of the class of `self`.   `self._shallow_copy([], name=name)` does not do that without a lot of modifications.  If you want me to do those modifications, I can look into that, but they might be pretty substantial.